File: tests/test_cases/gethostname.c
#include <stdio.h>
#include <unistd.h>

int main() 
{   
    int buffersize = 256;
    char hostbuffer[buffersize]; 
  
    gethostname(hostbuffer, buffersize);
  
    printf("Hostname: %s\n", hostbuffer); 

    return 0; 
} 

File: tests/test_cases/gethostname.py
#!/bin/python2
# Checks if gethostname.c runs the same natively as it does within Lind
# Lind's hostname is compared with "Lind" value, and native hostname is compared with socket.gethostname.
import re
import sys
import string
import socket

lind_results = sys.argv[1]
native_results = sys.argv[2]

# Deterministic line comparisons
if (socket.gethostname() != native_results.split()[1]):
    print "Hostname does not match in native!"
    exit(-1)

if (lind_results.split()[1]!="Lind"):
    print "Hostname does not match in Lind."
    exit(-1)